src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java

```diff
@@ -21,18 +23,29 @@
 @Repository
 public interface DirectoryElementRepository extends JpaRepository<DirectoryElementEntity, UUID> {
 
-    List<DirectoryElementEntity> findAllByParentId(UUID parentId);
+    List<DirectoryElementEntity> findAllByParentIdAndStashed(UUID parentId, boolean stashed);
 
-    @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId IS NULL AND d.type = 'DIRECTORY' AND (d.isPrivate=false or d.owner=:owner)")
+    List<DirectoryElementEntity> findAllByIdInAndStashed(List<UUID> uuids, boolean stashed);
+
+    @Modifying
+    @Transactional
+    @Query("DELETE FROM DirectoryElementEntity d WHERE d.id IN :elementsUuids")
+    void deleteAllById(List<UUID> elementsUuids);
```

Comment:
CrudRepository interface already contains a deleteAllById method

Reply:
Yes but with deleteAllByIdIn it will launch a query for each id. with @Query we will have only one query